libs/fluid-elements/tab-group/src/lib/tab-group.ts

```diff
@@ -51,50 +52,64 @@ export class FluidTabGroup extends LitElement {
   }
 
   /**
-   * Defines a tab to be active
+   * Defines the currently active tab id
    * @attr
    * @type string
    */
   @property({ type: String, reflect: true })
-  activetabid: string;
+  activeTabId: string;
+
+  /** Contains the tab id that is focus using ARROWKEYS and will be activated when the user presses SPACEBAR */
+  private _toBeActiveTab: string;
 
   /** Sets the active tab on click */
   private handleClick(event: FluidTabActivatedEvent): void {
-    if (this.activetabid != event.activeTab) {
-      this.activetabid = event.activeTab;
-      for (const tab of this.tabChildren) {
-        tab.active = tab.tabid === this.activetabid;
+    // Only continue when the clicked tab is different from the currently active tab
+    if (this.activeTabId != event.activeTab) {
+      // Resets the tabindexes
+      for (const tabItem of this.tabChildren) {
+        tabItem.tabindex = -1;
       }
-      this.dispatchEvent(new FluidTabGroupActiveTabChanged(this.activetabid));
+      this.activeTabId = event.activeTab;
+      // Contains the tab to be active
+      const tab = this.tabChildren.find(
+        (tabItem) => tabItem.tabid === event.activeTab,
+      )!;
+      tab.active = true;
+      tab.tabindex = 0;
+      this.dispatchEvent(new FluidTabGroupActiveTabChanged(this.activeTabId));
     }
   }
 
   /** Sets the active tab on keydown (ArrowLeft and ArrowRight to select / Enter and Space to confirm) */
-  private handleKeyDown(event: KeyboardEvent): void {
-    // Enter Space controll (validate selection)
+  private handleKeyUp(event: KeyboardEvent): void {
+    // Sets the focus outline when user tabbed into the fluid element
+    if (event.code === TAB) {
+      const tabbed = this.tabChildren.find((tab) => {
+        return tab.tabindex === 0;
+      });
+      tabbed!.tabbed = true;
```

Comment:
You cannot really rely on the fact, that you will find a currentlyFocussed tab. Please add a check if you actually found one.